internal/api/grpc/org/v2/query.go

```diff
@@ -0,0 +1,133 @@
+package org
+
+import (
+	"context"
+
+	"github.com/zitadel/zitadel/internal/api/grpc/object/v2"
+	"github.com/zitadel/zitadel/internal/domain"
+	"github.com/zitadel/zitadel/internal/query"
+	"github.com/zitadel/zitadel/internal/zerrors"
+	"github.com/zitadel/zitadel/pkg/grpc/org/v2"
+)
+
+func (s *Server) ListOrganizations(ctx context.Context, req *org.ListOrganizationsRequest) (*org.ListOrganizationsResponse, error) {
+	queries, err := listOrgRequestToModel(req)
+	if err != nil {
+		return nil, err
+	}
+	orgs, err := s.query.SearchOrgs(ctx, queries)
+	if err != nil {
+		return nil, err
+	}
+	orgs.RemoveNoPermission(ctx, s.checkPermission)
```

Comment:
why is that part of the API and not the SearchOrgs itself?

Reply:
Because of the permission check, which is not in the query, done like the permission check for the users.

Reply:
yes but each caller now needs to check that instead of having that as part of the query
IMO the query should only return the appropriate result (which should be done by the db in the future: https://github.com/zitadel/zitadel/discussions/7256)